tests/test_special_ops.py

```diff
@@ -532,3 +533,23 @@ def test_accuracy_stack(shape, dim, dtype):
     with flag_gems.use_gems():
         res_out = torch.stack(inp, dim)
     gems_assert_equal(res_out, ref_out)
+
+
+@pytest.mark.parametrize("shape", HSTACK_SHAPES)
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES + INT_DTYPES)
+def test_accuracy_hstack(shape, dtype):
+    if dtype in FLOAT_DTYPES:
+        inp = [torch.randn(s, dtype=dtype, device="cuda") for s in shape]
+    else:
+        inp = [
+            torch.randint(low=0, high=0x7FFF, size=s, dtype=dtype, device="cuda").to(
+                dtype
+            )
+            for s in shape
+        ]
+    ref_inp = [to_reference(_, True) for _ in inp]
```

Comment:
Since hstack involves no real computation, there is no need to upcast in `to_reference`.